src/main/scala/asciibot/asciibot.scala

```diff
@@ -0,0 +1,40 @@
+package asciibot
+
+import scala.io.Source
+
+import parser._
+
+import scala.collection.mutable.MutableList
+import picolib.maze.Maze
+import picolib.semantics._
+
+object AsciiBot extends App {
+  if (args.length != 2) {
+    println("Needs 2 arguments: maze file and program file")
+    System.exit(2)
+  }
+
+  val rules : List[Rule] = try {
+    AsciibotParser.parseLines(Source.fromFile(args(1)).getLines)
+  } catch {
+    case BlockError(m, sc, ec, sl, el) =>
+      println(s"Error: lines ${sl} - ${el} chars ${sc+1} - ${ec}: $m")
+      List()
+    case LineError(m,line) =>
+      println(s"Error: line ${line+1}: $m")
+      List()
+    case FileError(m) =>
+      println(s"Error: $m")
+      List()
```

Comment:
I wish we had better formatted error messages - right now we just throw exceptions and let Java treat them as runtime errors, but I wish we caught them in our program and just printed informative error messages and then exited.